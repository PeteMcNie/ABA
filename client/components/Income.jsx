import React from 'react'
import { connect } from 'react-redux'

import { sendIncomeToStore } from '../actions'

class Income extends React.Component {

  state = {
    incomes: [{ name: '', amount: '', frequency: 'weekly' }]
  }

  changeHandler = (evt, index) => {
    const { value, name } = evt.target
    const newIncomes = this.state.incomes
    newIncomes[index] = { ...newIncomes[index], [name]: value }
    this.setState({
      incomes: newIncomes
    })
  }

  submitHandler = (income, evt) => {
    evt.preventDefault()
    this.props.dispatch(sendIncomeToStore(income[0]))
  }

  render () {
    const { incomes } = this.state
    return (
      <>
        <form onSubmit={(evt) => this.submitHandler(incomes, evt)}>
          {
            incomes.map((incomeSource, index) => {
              return (
                <div key={index}>
                  <label htmlFor="name">Name </label>
                  <input type="text"
                    value={incomeSource.name}
                    placeholder="Income Type"
                    name="name"
                    onChange={evt => this.changeHandler(evt, index)} />

                  <label htmlFor="amount">Amount </label>
                  <input type="text"
                    value={incomeSource.amount}
                    placeholder="Amount"
                    name="amount"
                    onChange={evt => this.changeHandler(evt, index)} />
                  <button type="submit" value="submit">Submit</button>
                </div>
              )
            })
          }
        </form>
      </>
    )
  }
}

export default connect()(Income)
