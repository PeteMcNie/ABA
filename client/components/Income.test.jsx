import React from 'react'
import { Route } from 'react-router-dom'
import { renderWithRedux } from '../testing/utils'
import { screen } from '@testing-library/react'

import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'

import Income from './Income'

const initialState = {
  income: [
    {name: 'Wages', amount: '45000', occurrancy: 'weekly'}
  ]
}


test('Income has the correct structure', async () => {
  renderWithRedux(<Income />, {initialState})
  const form = await screen.findByText('Name')
  expect(form).toBeInTheDocument()
  expect(form).toMatchSnapshot()
})