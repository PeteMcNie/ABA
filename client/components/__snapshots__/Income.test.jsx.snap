// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Income has the correct structure 1`] = `
<label
  for="name"
>
  Name 
</label>
`;
